Victoria & Albert Museum.js

```diff
@@ -0,0 +1,198 @@
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes("/item/")) {
+		return "artwork";
+	}
+	else if (url.includes("/search/")) {
+		return "multiple";
+	}
+	return false;
+}
+
+function scrape(doc, url) {
+	var id = url.match(/item\/(O[0-9]+)/i)[1];
+
+	var apiUrl = "https://api.vam.ac.uk/v2/museumobject/" + id;
+	Zotero.Utilities.doGet(apiUrl, function (text) {
+		var museumobject = JSON.parse(text);
+		var data = museumobject.record;
+		var item = new Zotero.Item("artwork");
+		if (data.titles.length > 0) {
+			item.title = data.titles[0].title;
+			item.shortTitle = data.objectType;
+		}
+		else if (data.objectType) {
+			item.title = data.objectType;
+			item.shortTitle = data.objectType;
+		}
+
+		item.abstractNote = data.briefDescription;
+		item.medium = data.materialsAndTechniques;
+		item.artworkSize = data.dimensionsNote;
+		if (data.productionDates.length > 1) {
+			item.date = data.productionDates[0].date.text + " and further dates";
+		}
+		else if (data.productionDates.length > 0) {
+			item.date = data.productionDates[0].date.text;
+		}
+
+		item.archive = data.collectionCode.text;
+		item.libraryCatalog = 'Victoria & Albert Museum';
+		item.callNumber = data.accessionNumber;
+		item.rights = data.creditLine;
+		item.url = url;
+
+
+		var artistMakerPerson = data.artistMakerPerson;
+		for (var i = 0; i < artistMakerPerson.length; i++) {
+			// This is not ideal and assume surname, firstnames convention is correct
+			var fullName = artistMakerPerson[i].name.text;
+			var lastComma = fullName.lastIndexOf(",");
+			var firstName = "";
+			var lastName = "";
+
+			if (lastComma > 0) {
+				lastName = fullName.substring(0, lastComma);
+				firstName = fullName.substring(lastComma + 1);
+			}
+			else {
+				firstName = fullName;
+			}
+
+			item.creators.push({
+				lastName: lastName,
+				firstName: firstName,
+				creatorType: "author"
+			});
+		}
+
+		var artistMakerPeople = data.artistMakerPeople;
+		for (var j = 0; j < artistMakerPeople.length; j++) {
+			item.creators.push({
+				lastName: artistMakerPeople[j].name.text,
+				fieldMode: 1,
+				creatorType: "author"
+			});
+		}
+
+		var artistMakerOrganisations = data.artistMakerOrganisations;
+		for (var k = 0; k < artistMakerOrganisations.length; k++) {
+			item.creators.push({
+				lastName: artistMakerOrganisations[k].name.text,
+				fieldMode: 1,
+				creatorType: "author"
+			});
+		}
+
+		item.attachments.push({
+			url: url,
+			title: "Snapshot",
+			snapshot: true
+		});
+
+		item.complete();
+	});
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		var artworks = {};
+
+		var items = ZU.xpath(doc, '//figure[contains(@class, "b-object-card--etc")]//a');
+
+		if (items.length < 1) {
+			// TODO - other multiples
+			return false;
```

Comment:
You should move this `length` check into `detectWeb` for (much) better UX - as it stands, a search results page with no results will show the folder icon in the toolbar and then do absolutely nothing when the user clicks it.

See the web translator template (green plus menu in Scaffold) for an example of the pattern.